client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/post/index.tsx

```diff
@@ -14,7 +14,7 @@ export const getFirstPostPublished: TaskAction = ( task, flow, context ): Task =
 			( task.completed && isBlogOnboardingFlow( flow || null ) ) ||
 			false,
 		calypso_path: ! isBlogOnboardingFlow( flow || null )
```

Comment:
It feels like the logic to add the `origin` URL parameter could be done by building on the incoming `task.calypso_path` value.
```suggestion
		calypso_path: calypso_path: addQueryArgs( task.calypso_path, { origin: window.location.origin } );
```

The `origin` URL parameter should be ignored in most cases (though we should double-check!), and we can remove some conditional logic that is really unnecessary.

Reply:
I did a smoke test to see whether the writing flow works fine when there's an `origin` URL. It seems to ignore it, but it's hard to know what affect it might have with more extensive use. I'm not exactly sure what `origin` does, although I'm guessing it's something to do with making a flow work nicely regardless of whether it's a prod or dev environment.

I'm inclined to leave `origin` logic as it is for now (that is, only include it for some flows).
1. I can't be truly sure that it won't impact anything
2. It makes the URL a lot uglier. Not that the URL is super pretty without it, but it it includes an escaped `://` sequence

Reply:
FYI, the `origin` URL parameter is used in the following code:
https://github.com/Automattic/wp-calypso/blob/23e5d3ff396834274865eabc42e2cd1454547fa8/apps/wpcom-block-editor/src/wpcom/features/redirect-onboarding-user-after-publishing-post.js#L42-L66

We're discussing whether that component should be removed entirely in https://github.com/Automattic/wp-calypso/issues/94072 - I think the number of cases where it appears is relatively limited (though we might need some tracking to confirm that).

Reply:
I was worried that with groundskeeping going on at the same time as cleaning up flows that we'd end up in a situation where we feel unsure about whether some coding effort will be worth it 😄

I'm inclined to just fix the product as it currently is today, but then that'd leave us having to come back and removing the `origin` param as part of #94072